### ISSUE TITLE
Devenv needs to provide strong feedback when provisioning fails

### ISSUE THREAD

**Issue by ghaskins** [hyperledger-archives/fabric#2222]
## Description

Vagrant in general exposes a risk for confusion to users w.r.t. provisioning because it uses the same "vagrant ssh" plumbing as a normal user workflow would.  What this means is that the VM is created/booted until SSH is available.  Then the provisioner connects with ssh, runs its script until it either completes or fails, and finally exits out.  The user is then free to use "vagrant ssh" on their own, and they may or may not notice whether the provisioner script had errors.

We have a very poor mechanism in place for detecting this today:  We update the shell prompt via the PS1 environment variable as the very last step.  Therefore, we can tell in many bug reports that the user is running in a failed devenv simply by interpreting their prompt.  While helpful (to certain bug reviewers), this nuance is typically lost on the user.  They may, in fact, be running for a while in this incomplete environment with things not _quite_ right until they encounter some blocking issue.

What would be better is if we make this painfully obvious when the provisioning operation fails.  Here are some suggestions:
1. Add a MOTD early on (either as first provision step, or baseimage) that will display a very loud message on login that they should _not_ be seeing this message.  We can subsequently remove the MOTD as the last step after presumable success.
2. Similar to (1), but update the PS1 variable early on from its current "ubuntu-14.04" to "broken" or something along those lines.

**Comment by ghaskins** [comment]
@JonathanLevi

**Comment by ghaskins** [comment]
@srderson @binhn @christo4ferris Propose to target 0.5 branch.   If no objections, please label as such and I'll target the PR accordingly.

**Comment by srderson** [comment]
No objections from me

**Comment by JonathanLevi** [comment]
Yes, when it comes to error messages, it's much better to be "more direct" ;-)

As discussed on Slack and in view of some related questions/support requests, (@ghaskins et al.), I think that this is a great proposal and will be very useful (to target in the 0.5 branch as well).

Especially for new users, it is easy to miss some of the error message(s) when these occur, and then to (repeatedly) attempt/try to 'proceed' to the next step, not realizing that/where something went wrong.
